### PR TITLE
Remove native srcset experiment and update fallback logic

### PR DESCRIPTION
**Pull request by cathyxz**
This PR deletes the experiment specific code for `amp-img-native-srcset` so that we're running native `srcset` permanently instead of doing our own `srcset` logic (closes https://github.com/ampproject/amphtml/issues/11575). The experiment / prod/canary-config.json files now have to be edited as a separate PR per new build rules. Turns out that it broke the fallback image logic (which was dependent on checking whether the src should change on resize), so I refactored `updateImageSrc` into two listeners for on `load` and `error`. Annotations inline.

Testing: 
- [x] Chrome desktop
- [x] Safari desktop
- [x] FF desktop
- [x] Chrome Android
- [x] Chrome FF
- [x] Safari Iphone
- [x] Chrome Iphone
- [x] IE Windows

### In Summary
1. We don’t show image fallbacks after first load due to performance reasons, per https://github.com/ampproject/amphtml/pull/2575. 
2. Since we no longer manually set the src, whether or not we show a fallback, solely depends on whether the image loaded correctly (i.e. whether `loadPromise(this.img_)` rejects), no more checking for whether selected `src` equals `this.img_.src`. 
3. There is a problem with solely using `loadPromise` for this check. Namely, that there can be a race condition between the resize which calls `layoutCallback` and tries to check for error / load in order to toggle the fallback, and the load / error event itself. This causes problematic behavior on Firefox/Safari/Chrome where they don't show the fallback image. 
4. To solve this problem, I refactored the loadPromise then/catch block into two listeners for `load` and `error` on the underlying ``. 

### Not Addressed in this PR
1. Illogical behavior of fallbacks / placeholders per https://github.com/ampproject/amphtml/issues/10321 
2. Possibly revisiting the behavior of not toggling fallback images after first load. When resizing browsers, it seems like better UX to bring back the fallback image instead of showing a blank. But then again, I'm not sure what the performance concerns are, as well as how often people really play around with resizing browsers (or re-orienting their phones for that matter). Nonetheless, this is a bit of an edge case and will probably depend on actual usage. 

Further details on the different behaviors of `srcset` per browser at: go/amp-img-srcset

### CODE REVIEW

File: test/functional/test-amp-img.js [hunk 1]
```diff
@@ -317,7 +279,7 @@ describe('amp-img', () => {
       });
     });
 
-    it('should fallback only once', () => {
+    it('should fallback once and remove fallback once image loads', () => {
```
Comment (cathyxz) [inline test/functional/test-amp-img.js]: I combined this with the next test because the behavior here changed. Since we no longer set `src` based on the result of `parseSrcset`, the only thing that determines whether an image should show or hide its fallback is whether or not the image loaded, i.e. the result of `loadPromise`.

File: builtins/amp-img.js [hunk 2]
```diff
@@ -188,18 +139,13 @@ export class AmpImg extends BaseElement {
   /** @override */
   layoutCallback() {
     this.initialize_();
-    let promise = this.updateImageSrc_();
```
Comment (cathyxz) [inline builtins/amp-img.js]: This logic block was refactored into the `load` and `error` listeners on the right.

File: builtins/amp-img.js [hunk 3]
```diff
@@ -221,51 +167,33 @@ export class AmpImg extends BaseElement {
...
-          // The width should never be 0, but we fall back to the screen width
-          // just in case.
-          this.getViewport().getWidth() || this.win.screen.width,
-          this.getDpr());
-      if (src == this.img_.getAttribute('src')) {
-        return Promise.resolve();
-      }
-
-      this.img_.setAttribute('src', src);
+  hideFallbackImg_() {
```
Comment (cathyxz) [inline builtins/amp-img.js]: This logic is directly taken from the then block of then original `updateImageSrc` and should be unchanged.

File: builtins/amp-img.js [hunk 4]
```diff
@@ -221,51 +167,33 @@ export class AmpImg extends BaseElement {
...
    */
   onImgLoadingError_() {
-    this.getVsync().mutate(() => {
-      this.img_.classList.add('i-amphtml-ghost');
-      this.toggleFallback(true);
-      // Hide placeholders, as browsers that don't support webp
-      // Would show the placeholder underneath a transparent fallback
-      this.togglePlaceholder(false);
-    });
+    if (this.allowImgLoadFallback_) {
```
Comment (cathyxz) [inline builtins/amp-img.js]: Moved the if condition here to encompass the listener in a single function.

**Comment by cathyxz** [comment]
Based on the history on the original issue and the `<amp-img>` documentation, it seems like the main issue was that our documentation said we were "same as `srcset` attribute on the `img` tag", but our implementation was not. I've clarified that we have removed the polyfill (IE is the only browser that does not support native srcset at this point) and changed the wording from "AMP runtime" to "browser". The other main references to srcset that I can search for on our documentation website seem to be accurate descriptions of the browser spec (e.g. https://www.ampproject.org/docs/design/responsive/art_direction#srcset). 

@bpaduch are there other references to `amp-img` srcset that we need to update on our documentation? TLDR the change here is to remove our polyfill and propagate `srcset` on `amp-img`, so that `srcset` on `<amp-img>` will follow native browser behavior.

**Comment by cathyxz** [comment]
I think `<amp-anim>` also has a `srcset` polyfill. I can address that as a separate PR--the change should be very similar to what we did with `<amp-img>` here already.

File: test/manual/amp-img-fallback.amp.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      padding: 30px;
+      border: 10px solid red;
+    }
+  </style>
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+  <h1>amp-image</h1>
+  <!-- <h2>Unsupported formats</h2>
```
Comment (aghassemi) [inline test/manual/amp-img-fallback.amp.html]: uncomment

**Comment by cathyxz** [comment]
ping @bpaduch can we merge?

**Comment by jridgewell** [comment]
@cathyxz: Go ahead, we can review docs later.
